common/cgroup/cgroup.go

```diff
@@ -152,11 +153,17 @@ func getControllerRWFiles(controller string) []string {
 	return nil
 }
 
-func getOwnCgroupPath(controller string) (string, error) {
-	selfCgroupPath := "/proc/self/cgroup"
-	cg, err := os.Open(selfCgroupPath)
+func parseCgroupController(pid int, controller string) ([]string, error) {
+	var pidStr string
+	if pid == -1 {
+		pidStr = "self"
+	} else {
+		pidStr = strconv.Itoa(pid)
```

Comment:
Right, I think I was using a different pid in a previous iteration of this patch. Changed the function to parseOwnCgroupController and removed the parameter.